syntax = "proto3";

package protos;

message StudyType {
  message Value {
    string prediction = 1;
    float probability = 2;
    string model_name = 3;
  }
  repeated Value values = 1;
}

message InferenceData {
  oneof data {
    StudyType study_type = 1;
  }
}

message PaperInference {
  enum Provider {
    UNKNOWN_DERIVED_DATA_PROVIDER = 0;
    CONSENSUS = 1;
    SEMANTIC_SCHOLAR = 2;
  }

  enum Type {
    UNKNOWN_INFERENCE_TYPE = 0;
    STUDY_TYPE = 1;
  }

  int64 id = 1;

  int64 paper_id = 2;

  Type type = 3;

  Provider provider = 4;

  string source = 5;

  InferenceData data = 6;

  string created_by = 7;

  int64 created_at_usec = 8;
}
